OpticalCommunications-2023-N-robots-TCP/python_implementation/threads/Logger.py

```diff
@@ -0,0 +1,383 @@
+        disk = (total, available, used)
+
+        self.cursor.execute('''INSERT INTO diskTable (Total, Available, Used) VALUES (?, ?, ?)''', (disk))
+        
+        self.conn.commit()
+        
+    def interfacesData(self):
+        netCons = psutil.net_connections(kind='inet')
+        
+        for conn in netCons:
+            addrFamily = conn.family
+            localAddr = conn.laddr
+            remAddr = conn.raddr
+            addrType = conn.type
+            localAddrStr = ':'.join(str(x) for x in localAddr)
+            remAddrStr = ':'.join(str(x) for x in remAddr)
+            
+            tcpStatus = conn.status
+            network = (addrFamily, addrType, localAddrStr, remAddrStr, tcpStatus)
+            self.cursor.execute('''INSERT INTO interfacesTable (addressFamily, addressType, localAddress, remoteAddress, tcpStatus) VALUES (?, ?, ?, ?, ?)''',
+                        (network))
+
+            
+        self.conn.commit()
+
+    def processesData(self):
+        
+        process = psutil.process_iter()
+
+        for proc in process:
+            #pid = proc.pid()
+            procName = proc.name()
+            #gpuUsed = stats.processes[pid]['GPU']['usage']
+            cpuPercent = proc.cpu_percent()
+            memRss = proc.memory_info().rss / (1024 * 1024)  # Convert to MB
+            memVms = proc.memory_info().vms
+            memShared = proc.memory_info().shared
+            priority = proc.nice()
+            status = proc.status()
+            threads = proc.num_threads()
+        #gpuMemUsed = stats.processes[pid]['GPU']['memoryUsed'] / (1024 * 1024)  # Convert to MB
+        processes = (procName, cpuPercent, memRss, memVms, memShared, priority, status, threads)
+
+        self.cursor.execute('''INSERT INTO processesTable (processName, cpuPercent, memRss, memVms, memShared, Priority, Status, Threads) VALUES (?, ?, ?, ?, ?, ?, ?, ?)''',
+                (processes))
+        
+        self.conn.commit()
+        
+
+    def exportCsv(self, tableName, rows):
+        with open('{table_name}.csv', 'w', newline='', encoding='utf-8') as csv_file:
+            csvWriter = csv.writer(csv_file)
+            csvWriter.writerow([description[0] for description in self.cursor.description])  # Write headers automatically
+            csvWriter.writerows(rows)  
+
+        tableNames = ['addEvent', 'cpuTable', 'gpuTable', 'memRAMTable', 'memSWAPTable', 
+            'memEMCTable', 'memIRAMTable', 'engTable', 'sensorsTable', 'diskTable', 'interfaceTable', 'processesTable']
+
+        for tableName in tableNames:
+            self.cursor.execute("SELECT * FROM {tableName}")
+            rows = self.cursor.fetchall()
+            if rows:
+                self.exportCsv(tableName, rows)
+
+        self.conn.commit()
+
+    def __del__(self):
+        print("Commiting changes to Database and Deconstructing Logger")
+        self.__jetson.close()
+        # Close sql connection 
+        self.conn.close() 
```

Comment:
was having some issues with this when testing, maybe add try catch because connection close will fail if it was forcefully closed previously?